script/scripts/BOBPoolMigration.s.sol

```diff
@@ -0,0 +1,189 @@
+            )
+        );
+        vm.stopPrank();
+    }
+
+    function setLimits(address user, uint256 denominator) internal {
+        vm.startPrank(owner);
+        pool.setLimits({
+            _tier: 100,
+            _tvlCap: zkBobPoolCap / denominator,
+            _dailyDepositCap: zkBobDailyDepositCap / denominator,
+            _dailyWithdrawalCap: zkBobDailyWithdrawalCap / denominator,
+            _dailyUserDepositCap: zkBobDailyUserDepositCap / denominator,
+            _depositCap: zkBobDepositCap / denominator,
+            _dailyUserDirectDepositCap: zkBobDailyUserDirectDepositCap / denominator,
+            _directDepositCap: zkBobDirectDepositCap / denominator
+        });
+        address[] memory users = new address[](1);
+        users[0] = user;
+        pool.setUsersTier(100, users);
+        vm.stopPrank();
+    }
+
+    function getVerificationValues() internal returns (VerificationValues memory) {
+        uint256 snapshot = vm.snapshot();
+
+        uint256 prev_balance = IERC20(bob_addr).balanceOf(address(pool));
+        makeWithdrawal();
+        uint256 new_balance = IERC20(bob_addr).balanceOf(address(pool));
+        uint256 prev_withdrawal_diff = prev_balance - new_balance;
+
+        prev_balance = new_balance;
+        makeDeposit(false);
+        new_balance = IERC20(bob_addr).balanceOf(address(pool));
+        uint256 prev_deposit_diff = new_balance - prev_balance;
+
+        prev_balance = new_balance;
+        makeFeesWithdrawal();
+        uint256 prev_fees = prev_balance - IERC20(bob_addr).balanceOf(address(pool));
+
+        setLimits(relayer_addr, 1);
+        ZkBobAccounting.Limits memory limits = pool.getLimitsFor(relayer_addr);
+
+        vm.revertTo(snapshot);
+        return VerificationValues({
+            withdrawalDiff: prev_withdrawal_diff,
+            depositDiff: prev_deposit_diff,
+            fees: prev_fees,
+            limits: limits
+        });
+    }
+
+    function fakeUSDC() internal {
+        // the new implementation does not contain `require` to check that
+        // returned from ecrecover address is the same as the tokens originator.
+        //
+        // This is a workaround to use an existing deposit to verify the new pool
+        // logic.
+        address usdc_impl = deployCode("UChildAdministrableERC20.sol:UChildAdministrableERC20");
+        bytes memory code = usdc_impl.code;
+        vm.etch(IERCProxy(usdc_addr).implementation(), code);
```

Comment:
minor: seems like you can use `EIP1967Proxy` instead of `IERCProxy` here to achieve the same

Reply:
addressed in 384ac22